xonsh/base_shell.py

```diff
@@ -357,7 +357,10 @@ def singleline(self, **kwargs):
 
     def precmd(self, line):
         """Called just before execution of line."""
-        self.precwd = os.getcwd()
+        try:
+            self.precwd = os.getcwd()
+        except FileNotFoundError as f:
```

Comment:
the qa error is because of the unused variable `f` here. You can run the qa checks as 
`xonsh run-tests.xsh qa` locally.